### PR TITLE
Support for "volatile" keyword

### PR DESCRIPTION
**Pull request by jdemeyer**

### CODE REVIEW

**Comment by scoder** [comment]
From a code design perspective, it feels like there should be two separate subclasses for the `CConstOrVolatileType`, even if most of the implementation is in a shared base class. Also, I'm sure I'll fail to remember that "cv" stands for "const or volatile" when I look at the code later. And I don't see a real need for a flag that merges the two - why not just test both flags in conditions that apply to both?

**Comment by scoder** [comment]
Thinking about this some more, I'm unsure if `const` and `volatile` are really the same kind of thing from Cython's perspective. They are both C modifiers, ok, and `const` is certainly part of the type, but is `volatile` also part of the type? It feels more like a part of a variable (declaration) than a part of a data type.

**Comment by jdemeyer** [comment]
From a C/C++ point of view, "const" and "volatile" are often considers together. The name "cv" isn't my invention, it's regularly used when talking about const/volatile modifiers (just one example: http://en.cppreference.com/w/cpp/language/cv). But of course I can change `cv` to something else for Cython.

**Comment by jdemeyer** [comment]
> And I don't see a real need for a flag that merges the two - why not just test both flags in conditions that apply to both?

The problem is that in the current code, `is_const` is really used for two things: on the one hand, it is used for semantic const-ness (to prevent assignment for example). On the other hand, it is used as replacement for an `isinstance(..., CConstType)` check or a `hasttr(... , "const_base_type")` check. For the former, I am using `is_const`/`is_volatile` and for the latter I am using `is_const_or_volatile`.

**Comment by jdemeyer** [comment]
Is there any chance to revisit this?

File: Cython/Compiler/Parsing.py [hunk 1]
```diff
@@ -2481,14 +2481,25 @@ def p_c_simple_base_type(s, self_flag, nonempty, templates = None):
     pos = s.position()
     if not s.sy == 'IDENT':
         error(pos, "Expected an identifier, found '%s'" % s.sy)
-    if s.systring == 'const':
-        s.next()
+    if s.systring in ['const', 'volatile']:
```
Comment (scoder) [inline Cython/Compiler/Parsing.py]: Proposal:
```python
is_const = is_volatile = False
while s.sy == 'IDENT' and s.systring in ('const', 'volatile'):
    ...
if s.sy != 'IDENT':
    error(...)
if is_const or is_volatile:
    ...
```

**Comment by jdemeyer** [comment]
Fun fact: `cdef const const int a` is currently not an error, but I guess it should be :-)

**Comment by jdemeyer** [comment]
Why is Travis CI no longer testing this PR?

**Pull request merged**